include/cantera/thermo/PengRobinsonMFTP.h

```diff
@@ -0,0 +1,429 @@
+    virtual double densityCalc(double TKelvin, double pressure, int phase, double rhoguess);
+
+    virtual double densSpinodalLiquid() const;
+    virtual double densSpinodalGas() const;
+    virtual double pressureCalc(double TKelvin, double molarVol) const;
+    virtual double dpdVCalc(double TKelvin, double molarVol, double& presCalc) const;
+
+    //! Calculate dpdV and dpdT at the current conditions
+    /*!
+     *  These are stored internally.
+     */
+    void pressureDerivatives() const;
+
+    virtual void updateMixingExpressions();
+
+    //! Update the a and b parameters
+    /*!
+     *  The a and the b parameters depend on the mole fraction and the
+     *  temperature. This function updates the internal numbers based on the
+     *  state of the object.
+     */
+    void updateAB();
+
+    //! Calculate the a and the b parameters given the temperature
+    /*!
+     * This function doesn't change the internal state of the object, so it is a
+     * const function.  It does use the stored mole fractions in the object.
+     *
+     * @param temp  Temperature (TKelvin)
+     * @param aCalc (output)  Returns the a value
+     * @param bCalc (output)  Returns the b value.
+     */
+    void calculateAB(double temp, double& aCalc, double& bCalc, double& aAlpha) const;
+
+    // Special functions not inherited from MixtureFugacityTP
+
+    double daAlpha_dT() const;
+    double d2aAlpha_dT2() const;
+
+    void calcCriticalConditions(double a, double b,double& pc, double& tc, double& vc) const;
+
+    //! Solve the cubic equation of state
+    /*!
+     * The P-R equation of state may be solved via the following formula:
+     *
+     *     V**3 - V**2(RT/P - b)  - V(2bRT/P - \alpha a/P + 3*b*b) - (a \alpha b/p - b*b RT/P - b*b*b) = 0
+     *
+     * Returns the number of solutions found. If it only finds the liquid
+     * branch solution, it will return a -1 or a -2 instead of 1 or 2.  If it
+     * returns 0, then there is an error.
+     * The cubic equation is solved using Nickall's method (http://www.nickalls.org/dick/papers/maths/cubic1993.pdf)
+     */
+    int NicholsSolve(double TKelvin, double pres, double a, double b, double aAlpha,
+                     double Vroot[3]) const;
+
+protected:
+    //! Form of the temperature parameterization
+    /*!
+     *  0 = There is no temperature parameterization of a or b
+     *  1 = The a_ij parameter is a linear function of the temperature
+     */
+    int m_formTempParam;
+
+    //! Value of b in the equation of state
+    /*!
+     *  m_b_current is a function of the temperature and the mole fractions.
+     */
+    double m_b_current;
+
+    //! Value of a and alpha in the equation of state
+    /*!
+     *  m_aAlpha_current is a function of the temperature and the mole fractions. m_a_current depends only on the mole fractions.
+     */
+    double m_a_current;
+    double m_aAlpha_current;
+
+    vector_fp a_vec_Curr_;
```

Comment:
Each member variable should have a docstring. For vectors / arrays, describe the dimensions.

Reply:
I have added a comment to mention dimensions of these vectors.